### PR TITLE
fix(core): ensure proper install for git dependency using Yarn classic and npm

### PR DESCRIPTION
**Pull request by colincasey**
**What's the problem this PR addresses?**
When a git dependency is specified that doesn't exist in the yarn cache it will be:
* cloned to a temporary folder
* the target package manager used will be detected
* dependencies will be installed using the target package manager
* the target package manager will then execute the `pack` command to create a tarball that can be installed
  * if any `pack` lifecycle scripts exist (i.e.; `prepack`, `postpack`) these will also be executed 

If a `pack` lifecycle script depends on dev dependencies to be installed this process can fail in cases where environment variables like the following are present:
* `NODE_ENV=production` (npm and Yarn classic) 
* `NPM_CONFIG_PRODUCTION=true` (npm) 

This process of creating the installable tarball should guarantee that *all* dependencies are available and not allow environment variables to change what is installed.  

**How did you fix it?**
This can be fixed by removing the environment variables described above from the subprocess spawned to handle the Yarn classic and npm install steps when preparing an external project.

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: packages/yarnpkg-core/sources/scriptUtils.ts [hunk 1]
```diff
@@ -326,7 +326,10 @@ export async function prepareExternalProject(cwd: PortablePath, outputPath: Port
             // We can't use `npm ci` because some projects don't have npm
             // lockfiles that are up-to-date. Hopefully npm won't decide
             // to change the versions randomly.
-            const install = await execUtils.pipevp(`npm`, [`install`], {cwd, env, stdin, stdout, stderr, end: execUtils.EndStrategy.ErrorCode});
+            const install = await execUtils.pipevp(`npm`, [
+              `install`,
+              ...[`dev`, `prod`, `optional`, `peer`].map(include => `--include=${include}`),
```
Comment (merceyz) [inline packages/yarnpkg-core/sources/scriptUtils.ts:331]: I don't see any mentions of this flag in the 6.x docs so we should filter out `NODE_ENV` instead.

**Review by merceyz — CHANGES_REQUESTED** [review]
We'll also need to do this for Yarn classic as well since it has the same behaviour.

File: packages/acceptance-tests/pkg-tests-specs/sources/protocols/git.test.js [hunk 2]
```diff
@@ -149,6 +149,35 @@ describe(`Protocols`, () => {
       45000,
     );
 
+    test(
+      `it should guarantee that all dependencies will be installed when using npm to setup npm repositories`,
+      makeTemporaryEnv(
+        {
+          dependencies: {
+            [`has-prepack-npm`]: startPackageServer().then(url => `${url}/repositories/has-prepack-npm.git`),
```
Comment (merceyz) [inline packages/acceptance-tests/pkg-tests-specs/sources/protocols/git.test.js:157]: The other repositories has the package manager name first

File: packages/yarnpkg-core/sources/scriptUtils.ts (hunk 1 above)
Comment (colincasey) [reply packages/yarnpkg-core/sources/scriptUtils.ts]: 👍  done

**Comment by merceyz** [comment]
```
Rejected to value: {"externalException": {"message": "Invalid or unexpected token"}}
```

That error is happening because the content of the file is:
```js
'module.exports
=
42;'

```

**Comment by colincasey** [comment]
thanks @merceyz.  i had to add that output temporarily to understand why windows was failing with the syntax error.  next commit should sort things out 😅

**Review by merceyz — CHANGES_REQUESTED** [review]
Perfect, could you update the description to match the final state of the PR?

**Review by merceyz — APPROVED** [review]
LGTM, thanks!

**Pull request merged**